added option on http transport for having default attributes to be i… (#5876)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)
- Support for stdoutlog exporter in `go.opentelemetry.io/contrib/config`. (#5850)
- Add macOS ARM64 platform to the compatibility testing suite. (#5868)

> - Added option for adding default attributes to otel http transport in `go.opentelemetry.io/contrib/instrumentation/net/http/config`. (#5876)

Reviewer: ```suggestion
- Added option for adding default attributes to otel http transport in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`. (#5876)
```

- The `go.opentelemetry.io/contrib/config` package supports configuring `with_resource_constant_labels` for the prometheus exporter. (#5890)
- Add new runtime metrics to `go.opentelemetry.io/contrib/instrumentation/runtime`, which are still disabled by default. (#5870)
- Support for the `OTEL_HTTP_CLIENT_COMPATIBILITY_MODE=http/dup` environment variable in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` to emit attributes for both the v1.20.0 and v1.26.0 semantic conventions. (#5401)

> - Added option for extracting attributes from the http request in http transport in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`. (#5876)

Reviewer: Sorry for not pointing this out later. If the changelog entry mentions the new method, that's way easier for folks reading it to notice it.

```suggestion
- Add the `WithMetricsAttirbutesFn` option to allow setting dynamic, per-request metric attributes in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`. (#5876)
```

Author: np, i had to fix some conflicts anyway

- The `go.opentelemetry.io/contrib/bridges/otelzerolog` module.
  This module provides an OpenTelemetry logging bridge for `github.com/rs/zerolog`. (#5405)
- Add `WithGinFilter` filter parameter in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` to allow filtering requests with `*gin.Context`. (#5743)

> - Add the `WithMetricsAttirbutesFn` option to allow setting dynamic, per-request metric attributes in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`. (#5876)

Reviewer: ```suggestion
- Add the `WithMetricsAttributesFn` option to allow setting dynamic, per-request metric attributes in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`. (#5876)
```

Author: argh right didn't see the typo. Thanks!
